osu.Desktop/OsuGameDesktop.cs

```diff
@@ -117,9 +117,22 @@ private void fileDrop(object sender, FileDropEventArgs e)
         /// </summary>
         private class StableStorage : WindowsStorage
         {
+            private const string default_songs_path = "Songs";
+
+            private readonly string songsPath;
+            private bool usesCustomSongPath;
+            private readonly DesktopGameHost host;
+
+            public StableStorage(DesktopGameHost host)
+                : base(string.Empty, host)
+            {
+                this.host = host;
+                songsPath = locateSongsDirectory();
+            }
+
             protected override string LocateBasePath()
             {
-                static bool checkExists(string p) => Directory.Exists(Path.Combine(p, "Songs"));
+                static bool checkExists(string p) => File.Exists(Path.Combine(p, "osu!.exe"));
```

Comment:
This is a regression. osu! does not always locate its executable at the data path.

Reply:
ok alright. but checking for the Songs directory existence would fail in the case where the user changed his songs folder location and deletes the original one, right ?. So checking the existence of the stable installation by checking for the Songs directory presence is unreliable unless that directory is being recreated by the osu! stable client on every launch if it doesn't exist.  (in which case the old check can be restored)

Reply:
if you change the `BeatmapDirectory` to something else, it doesn't recreate the default folder

Reply:
In that case we should check for existence of files such as the user configuration file instead.